docs/open_source/testset_generation/rag_evaluation/index.md

```diff
@@ -141,6 +141,239 @@ Built-in metrics include `ragas_context_precision`, `ragas_faithfulness`, `ragas
 
 Alternatively, you can directly pass a list of answers instead of `get_answer_fn` to the `evaluate` function, you can then pass the retrieved documents as an optional argument `retrieved_documents` to compute the RAGAS metrics.
 
+### Custom Metrics
+
+**You can also implement your own metrics** using the base `Metric` class from Giskard.
+
+Here is an example of how you can implement a custom LLM-as-judge metric, as described in the [RAG cookbook](https://huggingface.co/learn/cookbook/en/rag_evaluation#evaluating-rag-performance) by Huggingface.
+
+#### Step 1 - Subclass the Metric class
+
+Implement the new metric in the `__call__` method: 
+
+```python
+
+from giskard.rag.metrics.base import Metric
+
+from giskard.llm.client import get_default_client
+from giskard.rag.question_generators.utils import parse_json_output
+from giskard.llm.errors import LLMGenerationError
+from giskard.rag.metrics.correctness import format_conversation
+
+
+class CorrectnessScoreMetric(Metric):
+
+    def __call__(self, question_sample: dict, answer: AgentAnswer) -> dict:
+        """ your docstring here
+        """
+
+        # Implement your LLM call with litellm
+        llm_client = self._llm_client or get_default_client()
+        try:
+            out = llm_client.complete(
+                messages=[
+                    ChatMessage(
+                        role="system",
+                        content=SYSTEM_PROMPT,
```

Comment:
I think it is fine to first define the sytem and input prompt above. For me personally, it makes it clearer.